### PR TITLE
flattening from sdk

### LINKED ISSUE #26: using SDK flattening logic in the plugin.
Using https://github.com/meltano/sdk/blob/main/singer_sdk/helpers/_flattening.py to do the flattening in the plugin.

### PR DESCRIPTION
**Pull request by prakharcode**
closes #26

### CODE REVIEW

**Comment by prakharcode** [comment]
This is a breaking change for the taps already using `flatten_records`. We can make it backward-compatible by using the same flag to trigger SDK flattening and issue a deprecation warning.

File: sample-config.json [hunk 1]
```diff
@@ -27,5 +27,6 @@
     "append_date_to_prefix_grain": "day",
     "append_date_to_filename": true,
     "append_date_to_filename_grain": "microsecond",
-    "flatten_records": false
+    "flattening_enabled": false,
```
Comment (crowemi) [inline sample-config.json:30]: @prakharcode are these properties automatically read in to the SDK? I noticed they aren't configured within the target itself `target.py`

Comment (prakharcode) [reply sample-config.json:30]: yes: https://github.com/search?q=repo%3Ameltano%2Fsdk%20get_flattening_options&type=code

**Review by crowemi — APPROVED** [review]

**Comment by crowemi** [comment]
> This is a breaking change for the implementations already using `flatten_records`. We can make it backward-compatible by using the same flag to trigger SDK flattening and issue a deprecation warning.

This would be ideal, one good thing on this is that we're versioning the target, hopefully people aren't using the `latest` build 🤷‍♂️

**Pull request merged**